lib/expand/join.go

```diff
@@ -462,6 +452,7 @@ func (p *Peer) submit(req *installpb.ExecuteRequest) bool {
 // execute executes either the complete operation or a single phase specified with req
 func (p *Peer) execute(req *installpb.ExecuteRequest) (dispatcher.Status, error) {
 	p.WithField("req", req).Info("Execute.")
+	p.startConnectLoop()
```

Comment:
What was the problem with having this in NewPeer? It's a bit confusing now that some peer routines are started there and this one here.

Reply:
Sorry - I've been meaning to add proper explanations.
The problem with connect loop starting in  `NewPeer` is that connect loop in its current form is a potentially mutating operation - i.e. it implicitly creates a cluster operation after successfully connecting to the controller. Since `NewPeer` is on the path of agent initialization, this can lead to undesired state if not properly controlled (which it wasn't when the agent could restart after a successfully completed operation resulting in bogus new and empty expand operations).

It also seemed to be clearer to move this behavior inside the part of code that handles client request to perform the operation thus making it explicit.